packages/app/src/cli/prompts/deploy-release.ts

```diff
@@ -0,0 +1,184 @@
+  ) {
+    infoTable.push(
+      configInfoTable.items.length === 0 ? {...configInfoTable, items: [{subdued: 'No changes'}]} : configInfoTable,
+    )
+  }
+  if (extensionsInfoTable) infoTable.push(extensionsInfoTable)
+  if (finalDeletedInfoTable.header !== '') infoTable.push(finalDeletedInfoTable)
+
+  const isDangerous = appTitle !== undefined && deletedInfoTable
+
+  if (isDangerous) {
+    confirmationResponse = await renderDangerousConfirmationPrompt({
+      message: question,
+      infoTable,
+      confirmation: appTitle,
+    })
+  } else {
+    let confirmationMessage
+
+    if (release) {
+      confirmationMessage = 'Yes, release this new version'
+    } else {
+      confirmationMessage = 'Yes, create this new version'
+    }
+
+    confirmationResponse = await renderConfirmationPrompt({
+      message: question,
+      infoTable,
+      confirmationMessage,
+      cancellationMessage: 'No, cancel',
+    })
+  }
+
+  const timeToConfirmOrCancelMs = new Date().valueOf() - timeBeforeConfirmationMs
+
+  await metadata.addPublicMetadata(() => ({
+    cmd_deploy_confirm_cancelled: !confirmationResponse,
+    cmd_deploy_confirm_time_to_complete_ms: timeBeforeConfirmationMs,
+  }))
+
+  return confirmationResponse
+}
+
+async function buildExtensionsContentPrompt(extensionsContentBreakdown: ExtensionIdentifiersBreakdown) {
+  const {fromDashboard, onlyRemote, toCreate: toCreateBreakdown, toUpdate} = extensionsContentBreakdown
+
+  let extensionsInfoTable
+  const extensionsInfo = [
+    ...toCreateBreakdown.map((identifier) => [identifier, {subdued: '(new)'}]),
+    ...toUpdate.map((identifier) => [identifier, {subdued: ''}]),
+    ...fromDashboard.map((identifier) => [identifier, {subdued: '(from Partner Dashboard)'}]),
+  ]
+  if (extensionsInfo.length > 0) {
+    extensionsInfoTable = {header: 'Extensions:', items: extensionsInfo}
+  }
+
+  let deletedInfoTable
+  const deleted = onlyRemote.map((field) => [{subdued: 'Extension:'}, field])
+  if (deleted.length > 0) {
+    deletedInfoTable = {
+      header: 'Removes:',
+      items: deleted,
+      bullet: '-',
+    }
+  }
+
+  await metadata.addPublicMetadata(() => ({
+    cmd_deploy_confirm_new_registrations: toCreateBreakdown.length,
+    cmd_deploy_confirm_updated_registrations: toUpdate.length,
+    cmd_deploy_confirm_removed_registrations: onlyRemote.length,
+  }))
+
+  return {extensionsInfoTable, deletedInfoTable}
+}
+
+async function buildConfigContentPrompt(
+  release: boolean,
+  configContentBreakdown?: ConfigExtensionIdentifiersBreakdown,
+) {
+  if (!configContentBreakdown)
+    return {
+      configInfoTable: {header: 'Configuration: ', items: []},
+      deletedInfoTable: undefined,
+    }
+
+  const {existingFieldNames, existingUpdatedFieldNames, newFieldNames, deletedFieldNames} = configContentBreakdown
+
+  const modifiedFieldNames = [
+    ...existingUpdatedFieldNames.map((field) => [field, {subdued: '(updated)'}]),
```

Comment:
There is a bug with `webhooks` always showing `(updated)` on the prompt table even when it has not been modified. This is due to `api_version` missing from the diff. This will now be included in the webhook module, so maybe we can include this now too

<img width="475" alt="image" src="https://github.com/Shopify/cli/assets/29992628/4ffb3d7e-e6f5-4866-8516-e92358e01ee3">


<img width="962" alt="image" src="https://github.com/Shopify/cli/assets/29992628/8e5d1a99-6f54-4e19-9594-557aa9ff45d7">

Reply:
semi related to this (I think), and this is more of a UX question potentially, but when you add declarative webhooks for the first time to your TOML, it says `webhooks (new)`. This is _technically_ not true as you will always have `api_version` under webhooks and that is already a mandatory field. What do you think of this? Should it just always say `updated`?

Reply:
I'm going to address the inclusion of the `api_version` as a versioned field in a new PR just to avoid doing a lot of `rebases`.

> semi related to this (I think), and this is more of a UX question potentially, but when you add declarative webhooks for the first time to your TOML, it says webhooks (new). This is technically not true as you will always have api_version under webhooks and that is already a mandatory field. What do you think of this? Should it just always say updated?

I will open a thread in the channel to discuss this topic. However from my point of view this will only happen the first time the user deploy the versioned config as after it the `breakdown info` will use the `registered config app modules`. The problem here is that adding the current api config to the logic to detect this scenario will make it really more complex to avoid this edge case